skbuild/cmaker.py

```diff
@@ -474,17 +474,42 @@ def check_for_bad_installs():
                     ("      " + _install) for _install in bad_installs)
             )))
 
-    def make(self, clargs=(), config="Release", source_dir=".", env=None):
+    def make(self, clargs=(), config="Release", source_dir=".",
+             target="install", env=None):
         """Calls the system-specific make program to compile code.
+
+           target: string
+               Target generated by CMake performing the install.
+               The target ``install`` is the default. It is always
+               generated by CMake and install all components.
+               But users might create a new target in CMake,
+               ``install-runtime``, which only install the desired
+               components, i.e Development, RuntimeLibraries, etc.
+               In the `install(...)` set of functions in
+               `CMakeLists.txt` use the optional argument
+               COMPONENT to set them.
+
         """
         clargs, config = pop_arg('--config', clargs, config)
+        clargs, target = pop_arg('--target', clargs, target)
         if not os.path.exists(CMAKE_BUILD_DIR()):
             raise SKBuildError(("CMake build folder ({}) does not exist. "
                                 "Did you forget to run configure before "
                                 "make?").format(CMAKE_BUILD_DIR()))
 
+        # The default target `install` depends on the builtin target: build-all
+        # This ensures that `make install` always trigger a build.
+        # However, this is not guaranteed by other targets, so we build the
+        # project first.
+        # There is a CMake PR to expose the build-all internal target:
+        # https://gitlab.kitware.com/cmake/cmake/-/issues/8438
```

Comment:
```
        # The default target `install` depends on the builtin target: build-all
        # This ensures that `make install` always trigger a build.
        # However, this is not guaranteed by other targets, so we build the
        # project first.
        # There is a CMake PR to expose the build-all internal target:
        # https://gitlab.kitware.com/cmake/cmake/-/issues/8438
```

->

```
        # Workaround CMake issue #8438 
        # See https://gitlab.kitware.com/cmake/cmake/-/issues/8438
        # Due to a limitation of CMake preventing from adding a dependency
        # on the "build-all" built-in target, we explicitly build the project first when
        # the install target is different from the default on.
```

Reply:
Consider also moving the regular build (without the install target) before the actual installation.